components/formats-gpl/src/loci/formats/in/LIFReader.java

```diff
@@ -127,6 +127,7 @@ private static ImmutableMap<String, Integer> createChannelPriorities() {
 
   /** Offsets to memory blocks, paired with their corresponding description. */
   private List<Long> offsets;
+  private transient List<String> offsetBlockIDs;
```

Comment:
While reading this PR, my understanding that there should be a one-to-one mapping between `offsets` and `offsetBlockIDs` which could have been represented as a `Map<String,Long>`.
Is that accurate? And this would be a larger change that would impact other locations in the reader?

Reply:
Yes, this could be represented as a `Map<String, Long>` instead of a pair of lists. I wasn't totally sure if this block ID approach would be usable initially, so was going for a minimal change that didn't require changing everywhere that touches `offsets`. Given that tests etc. are passing, I'm OK with spending a little more time here to switch to `Map<String, Long>`, if you think it's worthwhile.

Reply:
See what you think of d3683cedb1. I went back and forth a bit on whether `Map<String, Long>` really makes sense, and ultimately for the `openBytes` use case I think it doesn't. `openBytes` doesn't know or care about memory block IDs, as the offsets are meant to already be in the correct order according to series index. Switching `offsets` to a `List` of objects that captures both the memory block ID and actual offset seemed like a reasonable compromise, but happy to hear other thoughts.